pkg/plugins/resources/hcl/main.go

```diff
@@ -0,0 +1,156 @@
+package hcl
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/minamijoyo/hcledit/editor"
+	"github.com/mitchellh/mapstructure"
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/result"
+	"github.com/updatecli/updatecli/pkg/core/text"
+	"github.com/updatecli/updatecli/pkg/plugins/utils"
+)
+
+type Hcl struct {
+	spec             Spec
+	contentRetriever text.TextRetriever
+	files            map[string]file // map of file paths to file contents
+}
+
+type file struct {
+	originalFilePath string
+	filePath         string
+	content          string
+}
+
+func New(spec interface{}) (*Hcl, error) {
+	newSpec := Spec{}
+
+	err := mapstructure.Decode(spec, &newSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	newResource := &Hcl{
+		spec:             newSpec,
+		contentRetriever: &text.Text{},
+	}
+
+	err = newResource.spec.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	newResource.files = make(map[string]file)
+	// File as unique element of newResource.files
+	if len(newResource.spec.File) > 0 {
+		filePath := strings.TrimPrefix(newResource.spec.File, "file://")
+		newResource.files[filePath] = file{
+			originalFilePath: filePath,
+			filePath:         filePath,
+		}
+	}
+	// Files
+	for _, filePath := range newResource.spec.Files {
+		filePath := strings.TrimPrefix(filePath, "file://")
+		newResource.files[filePath] = file{
+			originalFilePath: filePath,
+			filePath:         filePath,
+		}
+	}
+
+	return newResource, nil
+}
+
+func (h *Hcl) Query(resourceFile file) (string, error) {
+	query := h.spec.Path
+
+	sink := editor.NewAttributeGetSink(query)
+
+	inStream := strings.NewReader(resourceFile.content)
+	outStream := new(bytes.Buffer)
+	err := editor.DeriveStream(inStream, outStream, resourceFile.filePath, sink)
+	if err != nil {
+		return "", err
+	}
+
+	// When a value isn't found hcledit returns an empty byte array,
+	// where as result always has \n appended so will be > 0
+	if outStream.Len() == 0 {
+		err := fmt.Errorf("%s cannot find value for path %q from file %q",
+			result.FAILURE,
+			query,
+			resourceFile.originalFilePath)
+		return "", err
+	}
+
+	queryOutput := outStream.String()
+	queryOutput = strings.TrimSpace(queryOutput)
+	queryOutput = strings.Trim(queryOutput, "\"")
+	return queryOutput, nil
+}
+
+func (h *Hcl) Apply(filePath string, valueToWrite string) error {
+	query := h.spec.Path
+
+	if _, err := strconv.Atoi(valueToWrite); err != nil {
+		valueToWrite = fmt.Sprintf(`"%s"`, valueToWrite)
+	}
```

Comment:
I've dealt with integers here, but this should possibly be gated behind bool, encase you wanted a string instead of int.

Reply:
well done, this sounds a similar problem than https://github.com/updatecli/updatecli/issues/1416